mne/beamformer/_compute_beamformer.py

```diff
@@ -44,14 +43,12 @@ def _check_src_type(filters):
     return filters, warn_text
 
 
-def _prepare_beamformer_input(info, forward, label, picks, pick_ori,
-                              fwd_norm=None, apply_proj=True):
-    """Input preparation common for all beamformer functions.
-
-    Check input values, prepare channel list and gain matrix. For documentation
-    of parameters, please refer to _apply_lcmv.
-    """
-    is_free_ori = forward['source_ori'] == FIFF.FIFFV_MNE_FREE_ORI
+def _prepare_beamformer_input(info, forward, label=None, pick_ori=None,
+                              noise_cov=None, rank=None, pca=False,
+                              combine_xyz='fro', exp=None, limit=None,
+                              allow_fixed_depth=True, limit_depth_chs=False):
+    """Input preparation common for LCMV, DICS, and RAP-MUSIC."""
```

Comment:
Do we expect to change the values of ``pca``, ``limit``, ``allow_fixed_depth``, ``limit_depth_chs`` ?
I wonder whether we need them available here or not.

Reply:
Oh, I see now part of this is handled by ``_check_depth`` in ``make_lcmv`` - not sure I understand fully how this interacts with setting defaults here, could you maybe quickly explain?

Reply:
The defaults here will be overridden by `make_lcmv`. In RAP-MUSIC they will not. In theory we could have no defaults for any of these but it makes the code longer in RAP-MUSIC (more invasive change, does not effectively keep old defaults, etc.). Six of one, half a dozen of the other...

Reply:
Note that we have a function ``combine_xyz`` from ``inverse`` in use in the beamformer code as well